src/test/java/org/wso2/securevault/secret/SecretManagerTest.java

```diff
@@ -0,0 +1,156 @@
+        Assert.assertEquals(2, actual.length);
+    }
+
+    @Test(description = "Test case for isPropValueValidated() method. ")
+    public void testIsPropValueValidated() throws Exception {
+
+        Properties configProps = getConfigProperties();
+        for (Map.Entry<Object, Object> entry : configProps.entrySet()) {
+            Boolean actual = Whitebox.invokeMethod(secretManager, "isPropValueValidated", entry.getValue());
+            Assert.assertTrue(actual);
+        }
+    }
+
+    @Test(description = "Test case for getPropertiesFromSecretConfigurations() method. ")
+    public void testGetPropertiesFromSecretConfigurations() throws Exception {
+
+        for (Map.Entry<Object, Object> entry : getConfigProperties().entrySet()) {
+            String actual = Whitebox.invokeMethod(secretManager, "getPropertiesFromSecretConfigurations",
+                    getConfigProperties(), entry.getKey().toString());
+            Assert.assertEquals(entry.getValue(), actual);
+        }
+    }
+
+    @Test(description = "Test case for resolveSecret() method. ")
+    public void testResolveSecretNovel() throws Exception {
+
+        Whitebox.invokeMethod(secretManager, "readLegacyProviders", getConfigProperties());
+        Whitebox.invokeMethod(secretManager, "readNovelProviders", getConfigProperties());
+        String alias = secretManager.resolveSecret("vault:hashicorp:admin_password");
+        Assert.assertEquals("admin_password", alias);
+    }
+
+    @Test(description = "Test case for resolveSecret() method. ")
+    public void testResolveSecretLegacy() throws Exception {
+
+        Whitebox.invokeMethod(secretManager, "readLegacyProviders", getConfigProperties());
+        Whitebox.invokeMethod(secretManager, "readNovelProviders", getConfigProperties());
+        String alias = secretManager.resolveSecret("admin_password");
+        Assert.assertEquals("admin_password", alias);
+    }
+
+    @Test(description = "Negative test case for resolveSecret() method. ")
+    public void testResolveSecretNegative() throws Exception {
+
+        try {
+            Whitebox.invokeMethod(secretManager, "readLegacyProviders", getConfigProperties());
+            Whitebox.invokeMethod(secretManager, "readNovelProviders", getConfigProperties());
+            secretManager.resolveSecret("vault:admin_password");
+            Assert.fail();
+        } catch (IllegalArgumentException ex) {
+            Assert.assertTrue(ex.getMessage().contains(
+                    "Invalid Annotation, The annotation expected to have [provider_type , repository_type , alias] but got [vault, admin_password]"));
+        }
+    }
+
+    private Properties getConfigProperties() {
+
+        Properties configProperties = new Properties();
+        configProperties.setProperty("keystore.identity.location",
+                "/home/sanduni/Downloads/Software Setups/wso2is-5.11.0-m28/repository/resources/security/wso2carbon" +
```

Comment:
This kind of location cannot be provided

Reply:
Check pleases where we use jks files for test cases we usually add them as test resources

Reply:
Corrected